src/js/game/hud/parts/debug_info.js

```diff
@@ -40,5 +78,7 @@ export class HUDDebugInfo extends BaseHUDPart {
             this.tickDurationElement.innerText =
                 "Tick Dur: " + round3Digits(this.root.dynamicTickrate.averageTickDuration) + "ms";
         }
+
+        this.updatePositions();
```

Comment:
Only call updatePositions if the overlay is visible, otherwise it slows down regular rendering